starter-code/js/stack.js

```diff
@@ -0,0 +1,28 @@
+$('document').ready(function () {
+
+  var stack1 = new StackDataStructure();
+
+  $("#addBtn").on('click', function () {
+    var input = $("input").val()
+    if (input.length) {
+      if (stack1.addElement(input) != "Overflow") {
+        $('ul').prepend(`<li>${input}</li>`);
+      } else if ($('li').html() === "OVERFLOW") {
+
+      } else {
+        $('ul').prepend(`<li>OVERFLOW</li>`);
+        $('li:eq(0)').addClass('flow');
+      }
+    } else {}
+  })
+
+  $("#takeBtn").on('click', function () {
+    if ($('ul li').length == 0) {
```

Comment:
The actions you take on the front-end should be driven by your data, not your html elements.   Every time you click your take button, you should first check first whether you are  not at risk of underflow before you change anything.  What happens now is that you are adding and removing the same 'underflow' element every time you click the button.